### DISCUSSION TITLE
How to polyfill buffer when using web3 Solana connector

### DISCUSSION THREAD

**Discussion by kleinpetr** [nuxt/nuxt#29978] [Questions]
Hi,

I am facing a build issue when I try to build for `cloudflare_pages` 

```
ERROR  Cannot resolve "unenv/runtime/node/buffer/index/" from "~/Projects/playground/nuxt-walletconnect/node_modules/.pnpm/@keystonehq+bc-ur-registry@0.5.5/node_modules/@keystonehq/bc-ur-registry/dist/index.js" and externals are not allowed!
```

here is minimal reproduction https://stackblitz.com/github/kleinpetr/nuxt-solana

I was trying to enable experimental `clientNodeCompat: true` but it didn't help
I was trying to polyfill node by this example https://stackoverflow.com/questions/77903800/vite-solana-wallet-adapter but it also didn't help 

I wouldn't need to know how I could get more details and how to think about the polyfilling.

Thanks for your ideas :pray:

**Comment by williamchong** [comment]
I just ran into this, and to provide some context, the issue was the way `buffer` is included in some libraries 
In `@keystonehq/bc-ur-registry/dist/index.js` they probably wrote something like `import { Buffer } from 'buffer/';` , which explicitly requires npm package `buffer` instead of node's `Buffer`
unenv couldn't understand this syntax, and just concated the tailing `/` to `unenv/runtime/node/buffer/index` when replacing buffer imports
I have no idea how to `fix` this except replacing it in the origin libraries though

**Comment by kleinpetr** [comment]
That's right, the workaround is to use explicit aliases for this imports, and optionally update types when using typescript

nuxt.config.ts
```ts
...
  alias: {
    'buffer/': 'buffer/',
    'string_decoder/': 'string_decoder/',
    'process/': 'process/',
    'pino-pretty': 'unenv/runtime/mock/proxy',
  },

  hooks: {
    'prepare:types' ({ tsConfig }) {
      tsConfig.compilerOptions ||= {}
      tsConfig.compilerOptions.paths ||= {}
      for (const key of ['buffer/', 'string_decoder/', 'process/'] as const) {
        // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
        delete tsConfig.compilerOptions.paths[key]
      }
    },
  },
...
```

Thanks to @danielroe :pray:
